models/user.go

```diff
@@ -0,0 +1,54 @@
+package models
+
+import (
+	"time"
+
+	"github.com/astaxie/beego/orm"
+	"github.com/astaxie/beego/validation"
+	"golang.org/x/crypto/bcrypt"
+)
+
+type User struct {
+	Id             int
+	Email          string `orm:"unique"`
+	HashedPassword string
+
+	CreatedAt time.Time `orm:"auto_now_add;type(datetime)"`
+	UpdatedAt time.Time `orm:"auto_now;type(datetime)"`
+}
+
+func init() {
+	orm.RegisterModel(new(User))
+}
+
+func (u *User) TableName() string {
+	return "users"
+}
+
+func CreateUser(user *User, password string) error {
+	o := orm.NewOrm()
+
+	hashedPassword, _ := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
+	user.HashedPassword = string([]byte(hashedPassword))
+
+	valid := validate(user, password)
```

Comment:
Should we validate before hash the password? There is no need to has the password if the password doesn't pass the validation.

Reply:
Hmm good thinking @gutakk 👍

Reply:
Fixed in https://github.com/andyduong1920/go_marketer/pull/3/commits/564e973f75097045b7d3b441ccba5eeb0b4037fb @gutakk